### PR TITLE
Event scrubber

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/3374

### CODE REVIEW

**Review by bitsandfoxes — CHANGES_REQUESTED** [review]
I really like this and I think this is a great feature and quality of life improvement!
It looks like the scrubber is doing quite a bit of work, I think it would make sense to add some benchmarks to it.

File: src/Sentry/Internal/EventScrubber.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    private void ScrubStringDictionary(IDictionary<string, string> dict)
+    {
+        foreach (var key in dict.Keys.ToList())
+        {
+            if (Denylist.Contains(key, StringComparer.InvariantCultureIgnoreCase))
+            {
+                dict[key] = ScrubbedText;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (jamescrosswell) [reply src/Sentry/Internal/EventScrubber.cs:70]: I tried that with a benchmark and it increased both the execution time and the allocations:

```diff
| Method                     | Mean     | Error    | StdDev   | Allocated |
|--------------------------- |---------:|---------:|---------:|----------:|
- | ScrubEvent_DefaultDenyList | 39.01 μs | 0.760 μs | 0.961 μs |   4.09 KB |
+ | ScrubEvent_DefaultDenyList | 54.09 μs | 1.165 μs | 3.418 μs |  15.39 KB |
```

**Comment by jamescrosswell** [comment]
> It looks like the scrubber is doing quite a bit of work, I think it would make sense to add some benchmarks to it.

Done... managed to get execution time down `39.01 μs` to `8.250 μs` and reduced allocations from `4.09 KB` to `2.44 KB`. 

The main improvement came from changing the type of DenyList from `List<string>` to `HashSet<string>`.

**Review by bitsandfoxes — APPROVED** [review]
This is awesome! Thanks for the benchmarks! Wow, quite the improvement!

**Pull request merged**